contrib/node/src/python/pants/contrib/node/subsystems/yarnpkg_distribution.py

```diff
@@ -4,15 +4,27 @@
 import logging
 
 from pants.binaries.binary_tool import NativeTool
+from pants.binaries.binary_util import BinaryToolUrlGenerator
 
 
 logger = logging.getLogger(__name__)
 
 
+class YarnReleaseUrlGenerator(BinaryToolUrlGenerator):
+
+  _DIST_URL_FMT = 'https://github.com/yarnpkg/yarn/releases/download/{version}/yarn-{version}.tar.gz'
+
+  def generate_urls(self, version, host_platform):
```

Comment:
Type hints appreciated while the code is still fresh.

Reply:
I'd like to come back to this - I ran into `mypy` errors elsewhere (`pants.util.objects` dependency) when trying to apply types to this change.